core/org.wso2.carbon.user.core/src/main/java/org/wso2/carbon/user/core/common/AbstractUserStoreManager.java

```diff
@@ -701,9 +1154,13 @@ private boolean authenticateInternal(String userName, Object credential, boolean
 
             // We are here due to two reason. Either there is no secondary UserStoreManager or no
             // domain name provided with user name.
-
             try {
                 // Let's authenticate with the primary UserStoreManager.
+                // Get the relevant userID for the given username.
+                if (isUniqueUserIdEnabled()) {
```

Comment:
we have to check the  isUniqueUserIdEnabled property in abstractUserStoreManager instance. Let's this approach is correct or not.